### PR TITLE
RAP-3: Google Search

### PR DESCRIPTION
**Pull request by curtishjr**
Initial commit of RAP-3 (Google Search) capability.   I also fixed a bug that kept exports from working properly.

Note: There's still one outstanding issue, which is that results are not displayed in the UI in the correct order.

### CODE REVIEW

File: core/lookups.py [hunk 1]
```diff
@@ -151,9 +154,47 @@ def lookup_ip_censys_https(ip):
...
+    else:
+        sifter = core.google.DomainSifter(domain)
+    result = list()
+    try:
+        for info in core.google.search(parameter, limit=limit, sifter=sifter):
+            result.append(info.to_dict())
+    except Exception:
+        # Something went wrong, most likely when querying Google.  There's nothing we can really do about it, so we will
+        # log the error and return an empty list
+        logger.exception("Unexpected error performing Google search")
```
Comment (bauerm) [inline core/lookups.py]: It may be helpful to log the exception text here as well

**Comment by bauerm** [comment]
+1

File: static/js/pivoteer.js [hunk 2]
```diff
@@ -214,6 +220,6 @@ $("#export").on( "click", "a", function(event) {
     var link = $(this);
     var linkURL = link.attr("href");
     var current_pivot = $("#pivot_navigator").find("li").last().find("span").text();
-    var fullURL = linkURL + "?indicator=" + current_pivot;
+    var fullURL = linkURL + "&indicator=" + current_pivot;
```
Comment (sbgoodm) [inline static/js/pivoteer.js]: is this change intentional? Is this a bug fix?

Comment (curtishjr) [inline static/js/pivoteer.js]: This is a bug fix.   Exports do not work properly without it.  That's because of how the links are built from pivoteer.html:
```
        <li class="pull-right">
            <div class="btn-group dropdown" id="export">
              <button class="btn btn-primary btn-md dropdown-toggle" type="button" data-toggle="dropdown" aria-haspopup="true" aria-expanded="true">
                  Export
                  <span class="glyphicon glyphicon-save" aria-hidden="true"></span>
              </button>
              <ul class="dropdown-menu" aria-labelledby="dropdownMenu1">
                <li><a href="{% url 'Pivoteer_Export' %}">All Records</a></li>
                <li><a href="{% url 'Pivoteer_Export' %}?filter=recent">Recent Records</a></li>
                <li><a href="{% url 'Pivoteer_Export' %}?filter=historical">Historical Records</a></li>
                <li><a href="{% url 'Pivoteer_Export' %}?filter=malware">Malware Records</a></li>
                <li><a href="{% url 'Pivoteer_Export' %}?filter=search">Search Records</a></li>
              </ul>
            </div>
        </li>
```

As a result, you would end up with a URL like `export?filter=search?indicator=domain.com` when what you actually needed was the `&`.

Comment (curtishjr) [inline static/js/pivoteer.js]: Though as I look at it now, I think my bug fix will fail for exporting *all* records, so I probably need to change it.  =(

Comment (curtishjr) [inline static/js/pivoteer.js]: Fixed in subsequent commit.
